partner_relations/models/res_partner.py

```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+# © 2013-2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import numbers
+from openerp import _, models, fields, exceptions, api
+from openerp.osv.expression import is_leaf, OR, FALSE_LEAF
+
+PADDING = 10
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    relation_count = fields.Integer(
+        'Relation Count',
+        compute="_compute_relation_count"
+    )
+
+    relation_ids = fields.One2many(
+        'res.partner.relation', string='Relations',
+        compute='_compute_relation_ids',
+        selectable=False,
+    )
+
+    relation_all_ids = fields.One2many(
+        'res.partner.relation.all', 'this_partner_id',
+        string='All relations with current partner',
+        auto_join=True, selectable=False, copy=False,
+    )
+
+    search_relation_id = fields.Many2one(
+        'res.partner.relation.type.selection', compute=lambda self: None,
+        search='_search_relation_id', string='Has relation of type',
+    )
+
+    search_relation_partner_id = fields.Many2one(
+        'res.partner', compute=lambda self: None,
+        search='_search_related_partner_id', string='Has relation with',
+    )
+
+    search_relation_date = fields.Date(
+        compute=lambda self: None, search='_search_relation_date',
+        string='Relation valid',
+    )
+
+    search_relation_partner_category_id = fields.Many2one(
+        'res.partner.category', compute=lambda self: None,
+        search='_search_related_partner_category_id',
+        string='Has relation with a partner in category',
+    )
+
+    @api.one
+    @api.depends("relation_ids")
+    def _compute_relation_count(self):
+        """Count the number of relations this partner has for Smart Button
+
+        Don't count inactive relations.
+        """
+        self.relation_count = len([r for r in self.relation_ids if r.active])
```

Comment:
Use `len(self.relation_ids.filtered('active'))`, but I think this is not even needed, as ORM by default hides non active records, so `len(self.relation_ids)` should be enough.